app/Http/Controllers/API/WpOrg/Plugins/Plugin_1_2_Controller.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Http\Controllers\API\WpOrg\Plugins;
+
+use App\Http\Controllers\Controller;
+use App\Http\Resources\PluginCollection;
+use App\Http\Resources\PluginResource;
+use App\Models\WpOrg\Plugin;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class Plugin_1_2_Controller extends Controller
+{
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function __invoke(Request $request): JsonResponse
+    {
+        $action = $request->query('action');
+        $slug   = $request->query('slug');
+
+        return match ($action) {
+            'query_plugins' => $this->queryPlugins($request),
+            'plugin_information' => $this->pluginInformation($slug),
+            default => response()->json(['error' => 'Invalid action'], 400)
+        };
+    }
+
+    private function pluginInformation(?string $slug): JsonResponse
+    {
+        if (!$slug) {
+            return response()->json(['error' => 'Slug is required'], 400);
+        }
+
+        $plugin = Plugin::query()->where('slug', $slug)->first();
+
+        if (!$plugin) {
+            return response()->json(['error' => 'Plugin not found'], 404);
+        }
+
+        return response()->json(new PluginResource($plugin));
+    }
+
+    private function queryPlugins(Request $request): JsonResponse
+    {
+        $page = max(1, (int) $request->query('page', '1'));
+        $perPage = (int) $request->query('per_page', '23');
```

Comment:
should default to 24 per page, that's the default for themes, and a quick query to upstream shows it's the case for plugins too.

Reply:
Fixed, changed to 24